src/main/java/io/mosip/mimoto/service/CredentialPDFGeneratorService.java

```
@@ -38,128 +38,92 @@
-                                }
-
-                                // Put the result into the rowProperties map
-                                rowProperties.put(originalKey, Map.of(nameFromDisplayObj, value));
-                            });
-
-                });
-
-        String qrCodeImage;
+
+        CredentialSupportedDisplayResponse firstDisplay = Optional.ofNullable(credentialsSupportedResponse.getDisplay())
+                .filter(list -> !list.isEmpty())
+                .map(List::getFirst)
+                .orElse(null);
+
+        String backgroundColor = firstDisplay != null ? firstDisplay.getBackgroundColor() : null;
+        String backgroundImage = firstDisplay != null && firstDisplay.getBackgroundImage() != null
+                ? firstDisplay.getBackgroundImage().getUri()
+                : null;
+        String textColor = firstDisplay != null ? firstDisplay.getTextColor() : null;
+        String credentialSupportedType = firstDisplay != null ? firstDisplay.getName() : null;
+
+        String face = extractFace(vcCredentialResponse);
+
+        displayProperties.forEach((key, valueMap) -> valueMap.forEach((display, val) -> {
+            String displayName = display.getName();
+            String locale = display.getLocale();
+            String strVal = formatValue(val, locale);
+            rowProperties.put(key, Map.of(displayName, strVal));
+        }));
+
+        String qrCodeImage = "";
         if (QRCodeType.OnlineSharing.equals(issuerDTO.getQr_code_type())) {
-            // Login flow
-            if (dataShareUrl.isEmpty()) {
-                qrCodeImage = constructQRCodeWithVCData(vcCredentialResponse);
-            } else {
-                qrCodeImage = constructQRCodeWithAuthorizeRequest(vcCredentialResponse, dataShareUrl);
-            }
+            qrCodeImage = constructQRCodeWithAuthorizeRequest(vcCredentialResponse, dataShareUrl);
```

can we have logic in `CredentialFormatHandler` to decide, do we support online sharing or not. If not then we can throw unsupported error